src/main/java/Parser.java

```diff
@@ -0,0 +1,110 @@
+import java.time.LocalDateTime;
+
+public class Parser {
+    /**
+     * Convert a String into a LocalDateTime that is understood by the computer.
+     *
+     * @param rawTimestamp time and date in string format.
+     * @return LocalDateTime with desired format.
+     * @throws DukeException if there is logic error or bad input.
+     */
+    public static LocalDateTime convertDateAndTime(String rawTimestamp) throws DukeException {
+        String[] dateTime = rawTimestamp.split(" ");
+        String[] datePortion = dateTime[0].split("/");
+        String timePortion = dateTime[1];
+
+        if (dateTime.length == 2) {
+            int date = Integer.parseInt(datePortion[0]);
+            int month = Integer.parseInt(datePortion[1]);
+            int year = Integer.parseInt(datePortion[2]);
+            int hour = Integer.parseInt(timePortion.substring(0, 2));
+            int minute = Integer.parseInt(timePortion) % 100;
+
+            return LocalDateTime.of(year, month, date, hour, minute);
+        } else {
+            throw new DukeException("Date time format invalid");
+        }
+    }
+
+    /**
+     * To deals with making sense of the user command.
+     *
+     * @param tasks the TaskList from the txt file.
+     * @param ui the message for interaction.
+     * @param inputText the text that user input.
+     * @param storage the location for retrieval and write the updated file.
+     */
+    public static void parse(TaskList tasks, Ui ui, String inputText, Storage storage) {
```

Comment:
It'd be good if `parse` merely parsed the input text, and the actual execution of the command were done elsewhere. (Single responsibility principle)